sdks/go/pkg/beam/runners/prism/internal/execute.go

```diff
@@ -86,33 +87,6 @@ func RunPipeline(j *jobservices.Job) {
 	j.Done()
 }
 
-// makeWorker creates a worker for that environment.
-func makeWorker(env string, j *jobservices.Job) (*worker.W, error) {
-	wk := worker.New(j.String()+"_"+env, env)
-
-	wk.EnvPb = j.Pipeline.GetComponents().GetEnvironments()[env]
-	wk.PipelineOptions = j.PipelineOptions()
-	wk.JobKey = j.JobKey()
-	wk.ArtifactEndpoint = j.ArtifactEndpoint()
-
-	go wk.Serve()
-
-	if err := runEnvironment(j.RootCtx, j, env, wk); err != nil {
-		return nil, fmt.Errorf("failed to start environment %v for job %v: %w", env, j, err)
-	}
-	// Check for connection succeeding after we've created the environment successfully.
-	timeout := 1 * time.Minute
-	time.AfterFunc(timeout, func() {
-		if wk.Connected() || wk.Stopped() {
-			return
-		}
-		err := fmt.Errorf("prism %v didn't get control connection to %v after %v", wk, wk.Endpoint(), timeout)
-		j.Failed(err)
-		j.CancelFn(err)
-	})
```

Comment:
We may still want this block of code moved to the MakeWorker method. This checks that the SDK has connected to the worker and serves as a liveness check for the SDK.

If we don't get this connection, the job *must* not hang forever. It's better that prism fails the job than waits forever, since it indicates there's something weird/bad going on in the SDK side.

Reply:
I added this back. Thank you.